code/modules/vehicles/mecha/equipment/weapons/weapons.dm

```diff
@@ -172,7 +172,7 @@
 	var/firing_angle = get_angle_with_scatter(chassis, current_target, projectile_to_fire.scatter, projectile_to_fire.p_x, projectile_to_fire.p_y)
 
 	playsound(chassis, fire_sound, 25, TRUE)
-	projectile_to_fire.fire_at(current_target, chassis, null, projectile_to_fire.ammo.max_range, projectile_to_fire.projectile_speed, firing_angle, suppress_light = HAS_TRAIT(src, TRAIT_GUN_SILENCED))
+	projectile_to_fire.fire_at(current_target, chassis, null, projectile_to_fire.ammo.max_range, projectile_to_fire.ammo.shell_speed, firing_angle, suppress_light = HAS_TRAIT(src, TRAIT_GUN_SILENCED))
```

Comment:
shell speed is already set for mech (and normal) guns in set gun modifiers or whatever its called, tanks should probs do the same (and get refactored to not be a funny copy of mecha guns)